nifi-nar-bundles/nifi-box-bundle/nifi-box-processors/src/main/java/org/apache/nifi/processors/box/ListBoxFiles.java

```diff
@@ -0,0 +1,262 @@
+    }
+
+    @Override
+    protected boolean isListingResetNecessary(final PropertyDescriptor property) {
+        return LISTING_STRATEGY.equals(property)
+            || FOLDER_ID.equals(property)
+            || RECURSIVE_SEARCH.equals(property);
+    }
+
+    @Override
+    protected Scope getStateScope(final PropertyContext context) {
+        return Scope.CLUSTER;
+    }
+
+    @Override
+    protected RecordSchema getRecordSchema() {
+        return BoxFileInfo.getRecordSchema();
+    }
+
+    @Override
+    protected String getDefaultTimePrecision() {
+        return PRECISION_SECONDS.getValue();
+    }
+
+    @Override
+    protected List<BoxFileInfo> performListing(
+        final ProcessContext context,
+        final Long minTimestamp,
+        final ListingMode listingMode
+    ) throws IOException {
+        final List<BoxFileInfo> listing = new ArrayList<>();
+
+        final String folderId = context.getProperty(FOLDER_ID).evaluateAttributeExpressions().getValue();
+        final Boolean recursive = context.getProperty(RECURSIVE_SEARCH).asBoolean();
+        final Long minAge = context.getProperty(MIN_AGE).asTimePeriod(TimeUnit.MILLISECONDS);
+
+
+        listFolder(listing, folderId, recursive, minAge);
+
+        return listing;
+    }
+
+    private void listFolder(List<BoxFileInfo> listing, String folderId, Boolean recursive, Long minAge) {
+        BoxFolder folder = getFolder(folderId);
+        for (BoxItem.Info itemInfo : folder.getChildren(
+            "id",
+            "name",
+            "item_status",
+            "size",
+            "created_at",
+            "modified_at",
+            "content_created_at",
+            "content_modified_at",
+            "path_collection"
+        )) {
+            if (itemInfo instanceof BoxFile.Info) {
+                BoxFile.Info info = (BoxFile.Info) itemInfo;
+
+                long createdAt = itemInfo.getCreatedAt().getTime();
+
+                if (Instant.now().toEpochMilli() - createdAt >= minAge) {
+                    BoxFileInfo boxFileInfo = new BoxFileInfo.Builder()
+                        .id(info.getID())
+                        .fileName(info.getName())
+                        .path("/" + info.getPathCollection().stream()
+                            .filter(pathItemInfo -> !pathItemInfo.getID().equals("0"))
+                            .map(BoxItem.Info::getName)
+                            .collect(Collectors.joining("/")))
+                        .size(info.getSize())
+                        .createdTime(info.getCreatedAt().getTime())
+                        .modifiedTime(info.getModifiedAt().getTime())
+                        .build();
+
+                    listing.add(boxFileInfo);
+                }
+            } else if (recursive && itemInfo instanceof BoxFolder.Info) {
+                BoxFolder.Info info = (BoxFolder.Info) itemInfo;
+                listFolder(listing, info.getID(), recursive, minAge);
+            }
+        }
+    }
```

Comment:
`Instant.now()` should be calculated once and before all network communications. Otherwise newer and newer files can be listed breaking the purpose of `minAge`.
As the method is recursive, the value should come from outside as a parameter. I'd suggest passing in `createdAtMax` instead of `minAge` which can be calculated as `Instant.now() - minAge`.
```suggestion
        long createdAtMax = Instant.now().toEpochMilli() - minAge;
        listFolder(listing, folderId, recursive, createdAtMax);

        return listing;
    }

    private void listFolder(List<BoxFileInfo> listing, String folderId, Boolean recursive, long createdAtMax) {
        BoxFolder folder = getFolder(folderId);
        for (BoxItem.Info itemInfo : folder.getChildren(
            "id",
            "name",
            "item_status",
            "size",
            "created_at",
            "modified_at",
            "content_created_at",
            "content_modified_at",
            "path_collection"
        )) {
            if (itemInfo instanceof BoxFile.Info) {
                BoxFile.Info info = (BoxFile.Info) itemInfo;

                long createdAt = itemInfo.getCreatedAt().getTime();

                if (createdAt <= createdAtMax) {
                    BoxFileInfo boxFileInfo = new BoxFileInfo.Builder()
                        .id(info.getID())
                        .fileName(info.getName())
                        .path("/" + info.getPathCollection().stream()
                            .filter(pathItemInfo -> !pathItemInfo.getID().equals("0"))
                            .map(BoxItem.Info::getName)
                            .collect(Collectors.joining("/")))
                        .size(info.getSize())
                        .createdTime(info.getCreatedAt().getTime())
                        .modifiedTime(info.getModifiedAt().getTime())
                        .build();

                    listing.add(boxFileInfo);
                }
            } else if (recursive && itemInfo instanceof BoxFolder.Info) {
                BoxFolder.Info info = (BoxFolder.Info) itemInfo;
                listFolder(listing, info.getID(), recursive, createdAtMax);
            }
        }
    }
```